deluge/common.py

```diff
@@ -961,23 +960,50 @@ def is_ipv6(ip):
 
     """
 
+    import socket
+
     try:
-        import ipaddress
-    except ImportError:
-        import socket
+        return socket.inet_pton(socket.AF_INET6, ip)
+    except OSError:
+        log.warning('Unable to verify IPv6 Address with socket.')
 
-        try:
-            return socket.inet_pton(socket.AF_INET6, ip)
-        except (OSError, AttributeError):
-            if windows_check():
-                log.warning('Unable to verify IPv6 Address on Windows.')
-                return True
-    else:
-        try:
-            return ipaddress.IPv6Address(decode_bytes(ip))
-        except ipaddress.AddressValueError:
-            pass
+    return False
+
+
+def is_net_interface(interface):
+    """A test to see if 'interface' is a valid network interface.
 
+    Args:
+        interface (str): The Interface to test. eg. eth0 linux. GUID on Windows.
+
+    Returns:
+        bool: Whether Interface is valid or not.
+
+    Examples:
+        >>> is_net_interface("eth0")
+        True
+        >>> is_net_interface("{7A30AE62-23ZA-3744-Z844-A5B042524871}")
+        True
+
+    """
+    if is_ipv4(interface) or is_ipv6(interface):
+        return False
+    try:
+        import ifaddr
+    except ImportError:
+        log.error('Unable to verify interfaces. Module ifaddr missing')
+        return False
+    adapters = ifaddr.get_adapters()
+    interface = interface.encode('utf-8')
```

Comment:
Alternatives? Cause it doesn't work when its not. So not sure what you mean by shouldn't be.

Reply:
So this should be `str` and it looks like there is platform bug in ifaddr 0.1.7 since it's returning `bytes` on Windows for adapter name, not good... https://github.com/pydron/ifaddr/pull/32

Can install that specific commit with:

```
pip install https://github.com/pydron/ifaddr/archive/37cb5334f392f12811d38d90ec891746e3247c76.zip
```